src/utils.c

```diff
@@ -424,3 +424,102 @@ SEXP startsWithAny(const SEXP x, const SEXP y, SEXP start) {
   return ScalarLogical(false);
 }
 
+SEXP frev(SEXP x, SEXP copyArg) {
+  if (INHERITS(x, char_dataframe) || INHERITS(x, char_datatable))
+    error(_("'x' should not be data.frame or data.table."));
+  if (!isNull(getAttrib(x, R_DimSymbol)))
+    error(_("'x' should not be matrix or array"));
+  if (!IS_TRUE_OR_FALSE(copyArg))
+    error(_("%s must be TRUE or FALSE."), "copy");
+  bool copy = LOGICAL(copyArg)[0];
+  int n = LENGTH(x);
```

Comment:
I think we should use `R_xlen_t n = xlength(x)` unless there's a good reason not to, especially as we use `uint64_t` for the loop counter. That's a data.table wide thing though, so not a necessary change.